examples/src/test/kotlin/AggregatesBuilderTest.kt

```diff
@@ -0,0 +1,939 @@
+                    )
+                )
+                // :snippet-end:
+        ))
+        assertEquals(3, docsStage.toList().size)
+    }
+
+    @Test
+    fun sampleTest() = runBlocking {
+        val sample = movieCollection.aggregate<Movie>(
+            listOf(
+                // :snippet-start: sample
+                Aggregates.sample(5)
+                // :snippet-end:
+            ))
+        assertEquals(5, sample.toList().size)
+    }
+
+    @Test
+    fun sortTest() = runBlocking {
+        val sort = movieCollection.aggregate<Movie>(
+            listOf(
+                // :snippet-start: sort
+                Aggregates.sort(
+                    Sorts.orderBy(
+                        Sorts.descending(Movie::year.name),
+                        Sorts.ascending(Movie::title.name)
+                    )
+                )
+                // :snippet-end:
+            )
+        )
+        assertEquals("The Dark Knight", sort.toList().first().title)
+    }
+
+    @Test
+    fun skipTest() = runBlocking {
+        val skip = movieCollection.aggregate<Movie>(listOf(
+            // :snippet-start: skip
+            Aggregates.skip(5)
+            // :snippet-end:
+        ))
+        assertEquals(4, skip.toList().size)
+    }
+
+    @Test
+    fun limitTest() = runBlocking {
+        val limit = movieCollection.aggregate<Movie>(listOf(
+            // :snippet-start: limit
+            Aggregates.limit(4)
+            // :snippet-end:
+        ))
+        assertEquals(4, limit.toList().size)
+    }
+
+    @Test
+    fun lookupTest() = runBlocking {
+        data class Comment(@BsonId val id: ObjectId, val name: String, val movie_id: Int, val text: String)
+        data class Results(@BsonId val id: Int, val title: String, val year: Int, val cast: List<String>, val genres: List<String>, val type: String, val rated: String, val plot: String, val fullplot: String, val runtime: Int, val imdb: Movie.IMDB, val joined_comments: List<Comment>)
+        val commentCollection = database.getCollection<Comment>("comments")
+
+        val comments = listOf(
+            Comment(id = ObjectId(), name = "John Doe", movie_id = 1, text = "This is a great movie. I enjoyed it a lot."),
+            Comment(id = ObjectId(), name = "Andrea Le", movie_id = 1, text = "Rem officiis eaque repellendus amet eos doloribus."),
+            Comment(id = ObjectId(), name = "Jane Doe", movie_id = 2, text = "I didn't like this movie. It was boring and predictable."),
+            Comment(id = ObjectId(), name = "Bob Smith", movie_id = 3, text = "This movie is a masterpiece. Highly recommended."))
+        commentCollection.insertMany(comments)
+
+        val lookup = movieCollection.aggregate<Document>(listOf(
+            // :snippet-start: lookup
+            Aggregates.lookup("comments", "_id", Comment::movie_id.name, Results::joined_comments.name
+            // :snippet-end:
+        )))
```

Comment:
b/c `Comment` and `Results` aren't in the example, think makes more sense to use string literals. also closing paren isn't in the snippet now:

```suggestion
            Aggregates.lookup("comments", "_id", "movie_id.name", "joined_comments")
            // :snippet-end:
        ))
```

Reply:
might also wanna wrap the snippet across multiple lines for readability

Reply:
Updated